hal/src/nRF52840/i2c_hal.cpp

```diff
@@ -674,3 +686,26 @@ int32_t HAL_I2C_Release(HAL_I2C_Interface i2c, void* reserved) {
     }
     return -1;
 }
+
+int HAL_I2C_Sleep(HAL_I2C_Interface i2c, bool sleep, void* reserved) {
+    if (i2c >= TOTAL_I2C) {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
+    }
+
+    I2cLock lk(i2c);
+    if (sleep) {
+        // Suspend I2C
+        CHECK_TRUE(m_i2c_map[i2c].enabled, SYSTEM_ERROR_NONE);
```

Comment:
Nit: sounds like having a `state` instead of two exclusive flags would look cleaner.

Reply:
Fixed, Introduced `hal_i2c_state_t`